packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -54,6 +58,82 @@ List<String> debugDescribeTransform(Matrix4 transform) {
   return matrix;
 }
 
+bool _extensionsInitialized = false;
+
+void initServiceExtensions() {
+  if (_extensionsInitialized)
+    return;
+
+  _extensionsInitialized = true;
+
+  assert(() {
+    registerExtension('flutter', _flutter);
+    // TODO: Expose debugDumpApp().
+    // TODO: Expose StatisticsOverlay.
+    registerExtension('flutter.debugPaint', _debugPaint);
+    registerExtension('flutter.timeDilation', _timeDilation);
+
+    // Emit an info level log message; this tells the debugger that the Flutter
+    // service extensions are registered.
+    log('Flutter initialized', name: 'flutter', level: 800);
+
+    return true;
+  });
+}
+
+/// Just respond to the request. Clients can use the existence of this call to
+/// know that the debug client is a Flutter app.
```

Comment:
I guess generally something using the `flutter` package, that would have access to these APIs, and for whom it would be useful to manipulate some of these debug levers.

It would be nice if there was one code place to register the flutter service extensions, and each layer could contribute whatever extensions were appropriate for itself. That way you don't have to have the rendering layer know about layers above it. I'm not sure how that would all be initialized however.